### ISSUE TITLE
Validation running loss

### ISSUE THREAD

**Issue by carmocca** [PyTorchLightning/pytorch-lightning#3231]
## 🚀 Feature
Trainer should compute a validation running loss just as it does for training.

### Motivation
I want to keep track of the running validation loss for each epoch. AFAIK this is not supported out-of-the-box by lightning.

### Pitch
`EvalResult` could include something similar to the `minimize` argument.

**Comment by rohitgr7** [comment]
if by running_loss you are referring to the a valid loss at each step then you can set `on_step=True` in `EvalResult().log('valid_loss', loss, on_step=True)`

**Comment by carmocca** [comment]
I would like to do the following:
```
TR - Epoch 0: 100%|█████████████████████████| 30/30 [00:06<00:00,  4.39it/s, loss={running_training_loss}]
VA - Epoch 0: 100%|█████████████████████████| 15/15 [00:01<00:00, 15.18it/s, loss={running_val_loss}]
TR - Epoch 1: 100%|█████████████████████████| 30/30 [00:07<00:00,  4.05it/s, loss={running_training_loss}]
VA - Epoch 1: 100%|█████████████████████████| 15/15 [00:01<00:00, 12.91it/s, loss={running_val_loss}]
```
so both the training and validation bars include running losses. The trainer already computes the `running_training_loss`.

**Comment by rohitgr7** [comment]
try: `EvalResult().log('valid_loss', loss, on_step=True, prog_bar=True)`

**Comment by carmocca** [comment]
I don't think that solves what I want. That just adds `step_valid_loss=200, epoch_valid_loss=147` (the last step and epoch validation losses) to the training progress bar.
What I want is the validation progress bar to include the running validation loss, which gets updated to the validation progress bar as the validation runs.

**Comment by awaelchli** [comment]
Same as #1239 ?

**Comment by carmocca** [comment]
Seems like it.

**Comment by carmocca** [comment]
@awaelchli Is this a feature you would merge?

**Comment by awaelchli** [comment]
Yes, see the linked issue, it is a todo feature. I think it is useful. If you have an idea how to implement it, go for it. Just be aware, there are some internal refactors happening at the moment all around Trainer, so might want to wait a week or so.
